From 5dacd8142bf8d93a198cb01fc7ef4896d4449254 Mon Sep 17 00:00:00 2001
Message-Id: <5dacd8142bf8d93a198cb01fc7ef4896d4449254.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 4 Oct 2014 22:24:23 +0530
Subject: [PATCH 48/75] MUST_REBASE [VPG]: adf/intel/vlv: Added dpst support
 while setting backlight

If dpst is enabled, we need to set user selected backlight via DPST :)

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I8c7346283a4cacbc0af9e70ff707822288ea6bd1
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    9 +++++++--
 drivers/video/adf/intel/core/vlv/vlv_dpst.c        |   15 +++++++++++++--
 2 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index de6a04f..40eb736 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -16,6 +16,7 @@
 #include <core/common/dsi/dsi_config.h>
 #include <core/common/intel_gen_backlight.h>
 #include <core/vlv/vlv_dc_config.h>
+#include <core/vlv/vlv_dpst.h>
 #include <intel_adf_device.h>
 #include "dsi_vbt.h"
 #include "intel_dsi.h"
@@ -60,8 +61,12 @@ static int dsi_set_brightness(struct intel_pipe *pipe, int level)
 	mutex_lock(&config->ctx_lock);
 	level = (level * 0xFF / BRIGHTNESS_MAX_LEVEL);
 
-	if (dsi_pipe->ops.set_brightness)
-		dsi_pipe->ops.set_brightness(level);
+	if (pipe->dpst_enabled)
+		vlv_dpst_set_brightness(pipe, level);
+	else {
+		if (dsi_pipe->ops.set_brightness)
+			dsi_pipe->ops.set_brightness(level);
+	}
 
 	panel = dsi_pipe->panel;
 	if (!panel || !panel->ops || !panel->ops->set_brightness) {
diff --git a/drivers/video/adf/intel/core/vlv/vlv_dpst.c b/drivers/video/adf/intel/core/vlv/vlv_dpst.c
index 7360eda..347cef4 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dpst.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dpst.c
@@ -57,13 +57,24 @@ static struct intel_dc_config *g_config;
 static u32
 vlv_get_backlight(struct intel_pipe *pipe)
 {
-	return 100;
+	struct dsi_pipe *dsi_pipe = to_dsi_pipe(pipe);
+	struct dsi_config *config = NULL;
+	struct dsi_context *ctx = NULL;
+
+	config = &dsi_pipe->config;
+	ctx = &config->ctx;
+
+	return ctx->backlight_level;
 }
 
 static void
 vlv_panel_set_backlight(struct intel_pipe *pipe, int level)
 {
-	pipe->ops->set_brightness(pipe, level);
+	struct dsi_pipe *dsi_pipe = to_dsi_pipe(pipe);
+	struct dsi_panel *panel = dsi_pipe->panel;
+
+	dsi_pipe->ops.set_brightness(level);
+	panel->ops->set_brightness(dsi_pipe, level);
 }
 
 static bool
-- 
1.7.9.5

